src/fu-engine-requirements.c

```diff
@@ -207,6 +208,36 @@ fu_engine_requirements_check_firmware(FuEngine *self,
 		}
 	}
 
+	/* check fwupd version requirement */
+	if (depth < 0) {
+		if (fu_version_compare(fwupd_version, "1.9.7", FWUPD_VERSION_FORMAT_UNKNOWN) < 0) {
+			g_set_error(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_NOT_SUPPORTED,
+				    "requirement 'child firmware' also needs %s >= 1.9.7",
+				    FWUPD_DBUS_SERVICE);
+			return FALSE;
+		}
+	} else if (depth == 0) {
+		if (fu_version_compare(fwupd_version, "1.6.1", FWUPD_VERSION_FORMAT_UNKNOWN) < 0) {
+			g_set_error(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_NOT_SUPPORTED,
+				    "requirement 'sibling firmware' also needs %s >= 1.6.1",
+				    FWUPD_DBUS_SERVICE);
+			return FALSE;
+		}
+	} else if (depth == 1) {
+		if (fu_version_compare(fwupd_version, "1.3.4", FWUPD_VERSION_FORMAT_UNKNOWN) < 0) {
+			g_set_error(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_NOT_SUPPORTED,
+				    "requirement 'parent firmware' also needs %s >= 1.3.4",
+				    FWUPD_DBUS_SERVICE);
```

Comment:
I don't really understand this.  Why have these checks for older fwupd versions?
The version of fwupd is already 1.9.10 when this commit goes in.  When we introduce a new check at 2.1 the client side code will be in the same commit as it.  Are you thinking to backport this entire commit (and similar ones in the future) to older stable trees?

I personally think this should really stay a server side check and it's unnecessary to check on client side too.

Reply:
Ahh, so no -- this isn't checking for older versions -- what this is doing is making sure that a new enough requirement for org.freedesktop.fwupd is set in the metainfo.xml file for each requirement type. This found one firmware which forgot the requirement -- although in that case the libgusb requirement was new enough to make it so nobody hit it. Essentially this is checking files either before they're uploaded to the LVFS or in cases where the LVFS isn't involved.

Reply:
i.e. if they're testing on fwupd 1.9.9 and then push to users and it doesn't work in 1.7.0 because the requirement type isn't supported. I think making it "not work" on 1.9.9 makes sure that vendors have to care less about when exactly we added things in fwupd -- if it works on a new fwupd it'll work everywhere.

Reply:
OH!  That makes a lot more sense now, thanks.